tpm2tools/signer.go

```diff
@@ -82,7 +88,10 @@ func (k *Key) GetSigner() (crypto.Signer, error) {
 	switch k.pubArea.Type {
 	case tpm2.AlgRSA:
 		sigScheme = k.pubArea.RSAParameters.Sign
-		sigAlg = tpm2.AlgRSASSA
+		sigAlg = sigScheme.Alg
+		if sigAlg != tpm2.AlgRSAPSS && sigAlg != tpm2.AlgRSASSA {
+			return nil, fmt.Errorf("unsupported signing algorithm: %v", sigAlg)
+		}
```

Comment:
This seems a little strange, maybe we should do the scheme error checking inside the switch. Something like:

```go
var sigScheme *tpm2.SigScheme
switch k.pubArea.Type {
case tpm2.AlgRSA:
	sigScheme = k.pubArea.RSAParameters.Sign
	if sigScheme.Alg != tpm2.AlgRSAPSS && sigScheme.Alg != tpm2.AlgRSASSA {
		return nil, fmt.Errorf("unsupported signing algorithm: %v", sigScheme.Alg)
	}
case tpm2.AlgECC:
	sigScheme = k.pubArea.ECCParameters.Sign
	if sigScheme.Alg != tpm2.AlgECDSA {
		return nil, fmt.Errorf("unsupported signing algorithm: %v", sigScheme.Alg)
	}
default:
	return nil, fmt.Errorf("unsupported key type: %v", k.pubArea.Type)
}

hash, err := sigScheme.Hash.Hash()
if err != nil {
	return nil, err
}
return &tpmSigner{k, hash}, nil
```

This makes the error checking clearer and shorter